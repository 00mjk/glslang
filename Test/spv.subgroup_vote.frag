#version 440

#extension GL_ARB_shader_group_vote : enable

flat in int idx;
out vec4 color;

void main() {
    bool b = (idx == 4);
    if (anyInvocationARB(b)) {
        color = vec4(1);
    }
    else if (allInvocationsARB(b)) {
        color = vec4(2);
    }
    else if (allInvocationsEqualARB(b)) {
        color = vec4(3);
    }
    else {
        color = vec4(4);
    }
}