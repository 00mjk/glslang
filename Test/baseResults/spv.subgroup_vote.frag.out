spv.subgroup_vote.frag
Warning, version 440 is not yet complete; most version-specific features are present, but some are missing.

Shader version: 440
Requested GL_ARB_shader_group_vote
0:? Sequence
0:8  Function Definition: main( (global void)
0:8    Function Parameters: 
0:9    Sequence
0:9      Sequence
0:9        move second child to first child (temp bool)
0:9          'b' (temp bool)
0:9          Compare Equal (temp bool)
0:9            'idx' (flat in int)
0:9            Constant:
0:9              4 (const int)
0:10      Test condition and select (temp void)
0:10        Condition
0:10        anyInvocation (global bool)
0:10          'b' (temp bool)
0:10        true case
0:11        Sequence
0:11          move second child to first child (temp 4-component vector of float)
0:11            'color' (out 4-component vector of float)
0:11            Constant:
0:11              1.000000
0:11              1.000000
0:11              1.000000
0:11              1.000000
0:10        false case
0:13        Test condition and select (temp void)
0:13          Condition
0:13          allInvocations (global bool)
0:13            'b' (temp bool)
0:13          true case
0:14          Sequence
0:14            move second child to first child (temp 4-component vector of float)
0:14              'color' (out 4-component vector of float)
0:14              Constant:
0:14                2.000000
0:14                2.000000
0:14                2.000000
0:14                2.000000
0:13          false case
0:16          Test condition and select (temp void)
0:16            Condition
0:16            allInvocationsEqual (global bool)
0:16              'b' (temp bool)
0:16            true case
0:17            Sequence
0:17              move second child to first child (temp 4-component vector of float)
0:17                'color' (out 4-component vector of float)
0:17                Constant:
0:17                  3.000000
0:17                  3.000000
0:17                  3.000000
0:17                  3.000000
0:16            false case
0:20            Sequence
0:20              move second child to first child (temp 4-component vector of float)
0:20                'color' (out 4-component vector of float)
0:20                Constant:
0:20                  4.000000
0:20                  4.000000
0:20                  4.000000
0:20                  4.000000
0:?   Linker Objects
0:?     'idx' (flat in int)
0:?     'color' (out 4-component vector of float)


Linked fragment stage:


Shader version: 440
Requested GL_ARB_shader_group_vote
0:? Sequence
0:8  Function Definition: main( (global void)
0:8    Function Parameters: 
0:9    Sequence
0:9      Sequence
0:9        move second child to first child (temp bool)
0:9          'b' (temp bool)
0:9          Compare Equal (temp bool)
0:9            'idx' (flat in int)
0:9            Constant:
0:9              4 (const int)
0:10      Test condition and select (temp void)
0:10        Condition
0:10        anyInvocation (global bool)
0:10          'b' (temp bool)
0:10        true case
0:11        Sequence
0:11          move second child to first child (temp 4-component vector of float)
0:11            'color' (out 4-component vector of float)
0:11            Constant:
0:11              1.000000
0:11              1.000000
0:11              1.000000
0:11              1.000000
0:10        false case
0:13        Test condition and select (temp void)
0:13          Condition
0:13          allInvocations (global bool)
0:13            'b' (temp bool)
0:13          true case
0:14          Sequence
0:14            move second child to first child (temp 4-component vector of float)
0:14              'color' (out 4-component vector of float)
0:14              Constant:
0:14                2.000000
0:14                2.000000
0:14                2.000000
0:14                2.000000
0:13          false case
0:16          Test condition and select (temp void)
0:16            Condition
0:16            allInvocationsEqual (global bool)
0:16              'b' (temp bool)
0:16            true case
0:17            Sequence
0:17              move second child to first child (temp 4-component vector of float)
0:17                'color' (out 4-component vector of float)
0:17                Constant:
0:17                  3.000000
0:17                  3.000000
0:17                  3.000000
0:17                  3.000000
0:16            false case
0:20            Sequence
0:20              move second child to first child (temp 4-component vector of float)
0:20                'color' (out 4-component vector of float)
0:20                Constant:
0:20                  4.000000
0:20                  4.000000
0:20                  4.000000
0:20                  4.000000
0:?   Linker Objects
0:?     'idx' (flat in int)
0:?     'color' (out 4-component vector of float)

